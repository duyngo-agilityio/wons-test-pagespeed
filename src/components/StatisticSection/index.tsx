// icons
import { FaHeart, FaShoppingBag } from 'react-icons/fa';
import { IoGameControllerSharp } from 'react-icons/io5';
import { MdWork } from 'react-icons/md';

// components
import { StatisticCard } from '@/components';

// models
import { IStatistic } from '@/models';

const StatisticSection = () => {
  const statisticData: IStatistic[] = [
    {
      value: '178+',
      label: 'Favorite Services',
      icon: <FaHeart className="h-6 w-6 text-blue-500" />,
      lightBgColor: 'bg-gray-50',
      darkBgColor: 'dark:bg-[#4a5463]',
    },
    {
      value: '20+',
      label: 'New Sales',
      icon: <IoGameControllerSharp className="h-6 w-6 text-teal-500" />,
      lightBgColor: 'bg-[#ceeeec]',
      darkBgColor: 'dark:bg-[#366263]',
    },
    {
      value: '190+',
      label: 'New Leads',
      icon: <FaShoppingBag className="h-6 w-6 text-pink-500" />,
      lightBgColor: 'bg-[#fff0f8]',
      darkBgColor: 'dark:bg-[#4a455c]',
    },
    {
      value: '12+',
      label: 'Referral',
      icon: <MdWork className="h-6 w-6 text-blue-500" />,
      lightBgColor: 'bg-gray-50',
      darkBgColor: 'dark:bg-[#4a5463]',
    },
  ];

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6">
      {statisticData.map((stat, index) => (
        <StatisticCard
          key={index}
          statistic={stat.value}
          label={stat.label}
          icon={stat.icon}
          lightBgColor={stat.lightBgColor}
          darkBgColor={stat.darkBgColor}
        />
      ))}
    </div>
  );
};

export default StatisticSection;
