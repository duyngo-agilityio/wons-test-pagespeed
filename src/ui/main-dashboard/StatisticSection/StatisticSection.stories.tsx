// Libs
import type { Meta, StoryObj } from '@storybook/react';

// components
import StatisticSection from './index';

const meta: Meta<typeof StatisticSection> = {
  title: 'Components/StatisticSection',
  component: StatisticSection,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof StatisticSection>;

export const Interactive: Story = {
  args: {},
};
