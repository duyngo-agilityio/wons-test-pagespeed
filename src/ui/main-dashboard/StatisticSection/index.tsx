// Icons
import { FaHeart, FaShoppingBag } from 'react-icons/fa';
import { IoGameControllerSharp } from 'react-icons/io5';
import { MdWork } from 'react-icons/md';

// Components
import { StatisticCard, NoProductAvailable } from '@/components';

// apis
import { getAllStatistics } from '@/api';

const StatisticSection = async () => {
  const { data: statisticsData } = await getAllStatistics();

  return (
    <div className="flex flex-col lg:flex-row">
      {statisticsData?.length ? (
        statisticsData
          .map((stat) => [
            {
              value: `${stat.favoriteServices}+`,
              label: 'Favorite',
              icon: <FaHeart className="h-5 w-5 text-[#605cf8]" />,
              lightBgColor: 'bg-gray-50',
              darkBgColor: 'dark:bg-[#4a5463]',
            },
            {
              value: `${stat.newSales}+`,
              label: 'New Sales',
              icon: <IoGameControllerSharp className="h-5 w-5 text-teal-500" />,
              lightBgColor: 'bg-[#ceeeec]',
              darkBgColor: 'dark:bg-[#366263]',
            },
            {
              value: `${stat.newLeads}+`,
              label: 'New Leads',
              icon: <FaShoppingBag className="h-5 w-5 text-pink-500" />,
              lightBgColor: 'bg-[#fff0f8]',
              darkBgColor: 'dark:bg-[#4a455c]',
            },
            {
              value: `${stat.referrals}+`,
              label: 'Referral',
              icon: <MdWork className="h-5 w-5 text-[#605cf8]" />,
              lightBgColor: 'bg-gray-50',
              darkBgColor: 'dark:bg-[#4a5463]',
            },
          ])
          .flat()
          .map((stat, index) => {
            const hasBorder = index < statisticsData.length - 1;
            return (
              <div key={index} className="w-full">
                <StatisticCard
                  statistic={stat.value}
                  label={stat.label}
                  icon={stat.icon}
                  lightBgColor={stat.lightBgColor}
                  darkBgColor={stat.darkBgColor}
                />
                {hasBorder && (
                  <div className="border-t border-gray-300 dark:border-gray-600 my-8" />
                )}
              </div>
            );
          })
      ) : (
        <NoProductAvailable message="No statistics available at the moment" />
      )}
    </div>
  );
};

export default StatisticSection;
