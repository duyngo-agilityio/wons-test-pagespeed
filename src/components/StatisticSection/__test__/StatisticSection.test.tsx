// components
import StatisticSection from '../index';

describe('StatisticSection component', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = testLibJestUtils.render(<StatisticSection />);
    expect(container).toMatchSnapshot();
  });
});
