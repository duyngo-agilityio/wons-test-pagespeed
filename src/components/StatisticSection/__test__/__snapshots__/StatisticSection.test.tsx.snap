// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticSection component renders correctly and matches snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
    >
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-large transition-transform-background motion-reduce:transition-none p-4 sm:p-6 m-2 bg-white shadow-lg rounded-10 dark:bg-gray-800 dark:shadow-none"
        style="border-radius: undefined;"
        tabindex="-1"
      >
        <div
          class="flex items-center flex-wrap"
        >
          <div
            class="mr-4 rounded-full flex items-center justify-center w-[63px] h-[63px] min-w-[50px] bg-gray-50 dark:bg-[#4a5463]"
            style="width: 63px; height: 63px; min-width: 50px;"
          >
            <svg
              class="h-6 w-6 text-blue-500"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              />
            </svg>
          </div>
          <div
            class="flex-1 min-w-0"
          >
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-bold text-[18px] sm:text-[24px] text-[#515c6a] dark:text-gray-850 truncate"
            >
              178+
            </p>
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-normal text-[16px] sm:text-[20px] text-blue-400 dark:text-gray-850  truncate"
            >
              Favorite Services
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-large transition-transform-background motion-reduce:transition-none p-4 sm:p-6 m-2 bg-white shadow-lg rounded-10 dark:bg-gray-800 dark:shadow-none"
        style="border-radius: undefined;"
        tabindex="-1"
      >
        <div
          class="flex items-center flex-wrap"
        >
          <div
            class="mr-4 rounded-full flex items-center justify-center w-[63px] h-[63px] min-w-[50px] bg-[#ceeeec] dark:bg-[#366263]"
            style="width: 63px; height: 63px; min-width: 50px;"
          >
            <svg
              class="h-6 w-6 text-teal-500"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M478.07 356.88 439 151c-8.86-40.35-23-71-88-71H145c-66 0-79.14 30.65-88 71L18 356.88c-7 34.12 4.43 61.25 33.37 71.81S103 423 119.18 391.3l15.42-30.52a16 16 0 0 1 14.28-8.78h198.28a16 16 0 0 1 14.28 8.78l15.42 30.52c16.14 31.7 38.88 48 67.81 37.39S485 391 478.07 356.88zM224 240h-48v48h-32v-48H96v-32h48v-48h32v48h48zm68 4a20 20 0 1 1 20-20 20 20 0 0 1-20 20zm44 44a20 20 0 1 1 20-20 20 20 0 0 1-20 20zm0-88a20 20 0 1 1 20-20 20 20 0 0 1-20 20zm44 44a20 20 0 1 1 20-20 20 20 0 0 1-20 20z"
              />
            </svg>
          </div>
          <div
            class="flex-1 min-w-0"
          >
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-bold text-[18px] sm:text-[24px] text-[#515c6a] dark:text-gray-850 truncate"
            >
              20+
            </p>
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-normal text-[16px] sm:text-[20px] text-blue-400 dark:text-gray-850  truncate"
            >
              New Sales
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-large transition-transform-background motion-reduce:transition-none p-4 sm:p-6 m-2 bg-white shadow-lg rounded-10 dark:bg-gray-800 dark:shadow-none"
        style="border-radius: undefined;"
        tabindex="-1"
      >
        <div
          class="flex items-center flex-wrap"
        >
          <div
            class="mr-4 rounded-full flex items-center justify-center w-[63px] h-[63px] min-w-[50px] bg-[#fff0f8] dark:bg-[#4a455c]"
            style="width: 63px; height: 63px; min-width: 50px;"
          >
            <svg
              class="h-6 w-6 text-pink-500"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
              />
            </svg>
          </div>
          <div
            class="flex-1 min-w-0"
          >
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-bold text-[18px] sm:text-[24px] text-[#515c6a] dark:text-gray-850 truncate"
            >
              190+
            </p>
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-normal text-[16px] sm:text-[20px] text-blue-400 dark:text-gray-850  truncate"
            >
              New Leads
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-large transition-transform-background motion-reduce:transition-none p-4 sm:p-6 m-2 bg-white shadow-lg rounded-10 dark:bg-gray-800 dark:shadow-none"
        style="border-radius: undefined;"
        tabindex="-1"
      >
        <div
          class="flex items-center flex-wrap"
        >
          <div
            class="mr-4 rounded-full flex items-center justify-center w-[63px] h-[63px] min-w-[50px] bg-gray-50 dark:bg-[#4a5463]"
            style="width: 63px; height: 63px; min-width: 50px;"
          >
            <svg
              class="h-6 w-6 text-blue-500"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
              />
            </svg>
          </div>
          <div
            class="flex-1 min-w-0"
          >
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-bold text-[18px] sm:text-[24px] text-[#515c6a] dark:text-gray-850 truncate"
            >
              12+
            </p>
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md font-normal text-[16px] sm:text-[20px] text-blue-400 dark:text-gray-850  truncate"
            >
              Referral
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
